In [11]:
import pandas as pd
import matplotlib as plt
address_data = pd.read_csv('address.csv', index_col = 0)
address_data.head()

address  address2  district  city_id  postal_code  \
address_id                                                                   
1              47 MySakila Drive       NaN   Alberta      300          NaN   
2             28 MySQL Boulevard       NaN       QLD      576          NaN   
3              23 Workhaven Lane       NaN   Alberta      300          NaN   
4           1411 Lillydale Drive       NaN       QLD      576          NaN   
5                 1913 Hanoi Way       NaN  Nagasaki      463      35200.0   

                   phone location          last_update  
address_id                                              
1                    NaN      ...  2014-09-25 21:30:27  
2                    NaN      ...  2014-09-25 21:30:09  
3           1.403334e+10      ...  2014-09-25 21:30:27  
4           6.172236e+09      ...  2014-09-25 21:30:09  
5           2.830338e+10      ...  2014-09-25 21:31:53

In [12]:
address_data.shape

(603, 8)

In [19]:
address_data.describe(include='all')
#address_data.describe()

address      district     city_id   postal_code  \
count                    603           600  603.000000    599.000000   
unique                   603           377         NaN           NaN   
top     770 Bydgoszcz Avenue  Buenos Aires         NaN           NaN   
freq                       1            10         NaN           NaN   
mean                     NaN           NaN  300.525705  50222.313856   
std                      NaN           NaN  173.600949  28986.085055   
min                      NaN           NaN    1.000000      3.000000   
25%                      NaN           NaN  150.500000  25182.500000   
50%                      NaN           NaN  300.000000  50805.000000   
75%                      NaN           NaN  450.500000  74920.000000   
max                      NaN           NaN  600.000000  99865.000000   

               phone location          last_update  
count   6.010000e+02      603                  603  
unique           NaN        1                  247  
top              NaN      ...  2014-09-25 21:31:45  
freq             NaN      603                    4  
mean    4.969694e+11      NaN                  NaN  
std     3.023514e+11      NaN                  NaN  
min     8.925233e+08      NaN                  NaN  
25%     2.112563e+11      NaN                  NaN  
50%     4.912714e+11      NaN                  NaN  
75%     7.623618e+11      NaN                  NaN  
max     9.980098e+11      NaN                  NaN

There are 603 records.<br> None supply a second address, <b>hence this field can be dropped</b>. <br>There are 3 missing values for district, 4 missing values for postal_code, four phone numbers were not provided.<br>
These missing data are not critical for the data interpretation.

In [16]:
# Let's remove the address2 since idt does not contain data
address_data.drop("address2", axis=1, inplace=True)

In [18]:
address_data.head()

In [22]:
# Let's group by district and identify district with highest records
address_data.groupby(address_data['district'])